services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,124 @@
+package org.openkilda.functionaltests.spec.northbound.switches
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.messaging.info.meter.SwitchMeterEntries
+import org.openkilda.testing.Constants
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.service.northbound.NorthboundService
+
+import org.junit.Ignore
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Issue
+
+class MetersSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    NorthboundService northboundService
+
+    def "Unable to delete meter with invalid ID"() {
+        given: "A switch"
+        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+
+        when: "Try to delete meter with invalid ID"
+        northboundService.deleteMeter(sw.getDpId(), -1)
+
+        then: "Got BadRequest because meter ID is invalid"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+    }
+
+    def "Default meters should be present on all switches"() {
+        given: "All Openflow 1.3 compatible switches"
+        def switches = topology.getActiveSwitches().findAll { it.ofVersion == "OF_13" }
+
+        when: "Try to get meters from all the switches"
+        List<SwitchMeterEntries> meters = switches.collect {
+            northboundService.getAllMeters(it.dpId)
+        }
+
+        then: "Only the default meters should be present on each switch"
+        meters.each {
+            assert it.meterEntries.size() == 2
+            assert it.meterEntries.every { it.rate == 200 }
+            assert it.meterEntries.every { it.burstSize == 0 }
+            assert it.meterEntries.every { it.flags.size() == 2 }
+            assert it.meterEntries.every { ["PKTPS", "STATS"].containsAll(it.flags) }
```

Comment:
Centecs don't support PKTPS, so we install a meter with KBPS and rate calculated using following formula: "(ratePkts * config.getDiscoPacketSize()) / 1024L"

Reply:
btw, if I'm not mistaken we are about to add "burst" flag here (at least such changes should be released soon)